lib/galaxy/managers/datasets.py

```diff
@@ -691,7 +691,7 @@ def deserialize_datatype(self, item, key, val, **context):
         if not self.app.datatypes_registry.get_datatype_by_extension(val).allow_datatype_change:
             raise exceptions.RequestParameterInvalidException("The target datatype does not allow datatype changes.")
         if not DatasetAssociationManager(self.app).ok_to_edit_metadata(item.dataset_id):
-            raise Exception("Dataset metadata could not be updated because it is used as input or output of a running job.")
+            raise exceptions.RequestParameterInvalidException("Dataset metadata could not be updated because it is used as input or output of a running job.")
```

Comment:
I changed this back as suggested - not sure if you and @nsoranzo finished arguing it out though.